cmd/account/generate-secret.go

```diff
@@ -0,0 +1,204 @@
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/sts"
+	awsv1alpha1 "github.com/openshift/aws-account-operator/pkg/apis/aws/v1alpha1"
+	"github.com/spf13/cobra"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/osd-utils-cli/cmd/common"
+	"github.com/openshift/osd-utils-cli/pkg/k8s"
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+)
+
+// newCmdGenerateSecret implements the generate-secret command which generates an new set of IAM User credentials
+func newCmdGenerateSecret(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	ops := newGenerateSecretOptions(streams, flags)
+	generateSecretCmd := &cobra.Command{
+		Use:               "generate-secret <IAM User name>",
+		Short:             "Generate IAM credentials secret",
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+
+	generateSecretCmd.Flags().StringVar(&ops.accountNamespace, "account-namespace", common.AWSAccountNamespace,
+		"The namespace to keep AWS accounts. The default value is aws-account-operator.")
+	generateSecretCmd.Flags().StringVarP(&ops.accountID, "account-id", "i", "", "AWS Account ID")
+	generateSecretCmd.Flags().StringVarP(&ops.profile, "aws-profile", "p", "", "specify AWS profile")
+	generateSecretCmd.Flags().StringVarP(&ops.region, "aws-region", "r", common.DefaultRegion, "Specify AWS region")
+	generateSecretCmd.Flags().StringVar(&ops.secretName, "secret-name", "", "Specify name of the generated secret")
+	generateSecretCmd.Flags().StringVar(&ops.secretNamespace, "secret-namespace", "aws-account-operator", "Specify namespace of the generated secret")
+	generateSecretCmd.Flags().BoolVar(&ops.printSecret, "quiet", false, "Suppress logged output")
+
+	return generateSecretCmd
+}
+
+// generateSecretOptions defines the struct for running generate-secret command
+type generateSecretOptions struct {
+	accountName      string
+	accountID        string
+	accountNamespace string
+	iamUsername      string
+
+	secretName      string
+	secretNamespace string
+	printSecret     bool
+	outputPath      string
+
+	// AWS config
+	region  string
+	profile string
+	cfgFile string
+
+	flags *genericclioptions.ConfigFlags
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+func newGenerateSecretOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags) *generateSecretOptions {
+	return &generateSecretOptions{
+		flags:     flags,
+		IOStreams: streams,
+	}
+}
+
+func (o *generateSecretOptions) complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "IAM User name argument is required")
+	}
+	o.iamUsername = args[0]
```

Comment:
Should we set a default value for the username, instead of getting it via cli parameter?

Reply:
Since this is a more generic option, I think we should make it required. Otherwise, someone could accidentally delete and create an AccessKey for the default user without meaning to.